cookbooks/aws-parallelcluster-slurm/files/default/head_node_slurm/slurm/pcluster_fleet_config_generator.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import argparse
+import copy
+import json
+import logging
+
+import yaml
+
+log = logging.getLogger()
+
+
+class CriticalError(Exception):
+    """Critical error for the script."""
+
+    pass
+
+
+def generate_fleet_config_file(output_file, input_file):
+    """
+    Generate configuration file used by Fleet Manager in node daemon package.
+
+    Generate fleet-config.json
+    {
+        "my-create-fleet-queue": {
+            "my-compute-resource": {
+                "CapacityType": "on-demand|spot",
+                "AllocationStrategy": "lowest-price"
+                "InstanceTypeList": [
+                    { "InstanceType": ... }
+                ],
+                "MaxPrice": ...
+            }
+        },
+        "my-run-instances-queue": {
+            "my-compute-resource": {
+                "CapacityType": "on-demand|spot",
+                "AllocationStrategy": "lowest-price"
+                "InstanceType": ...,
```

Comment:
in the definition of this interface would it make sense to only use InstanceTypeLIst and add another fields which makes it explicit what the API to use is, whether run_instances or fleet? So that 1. we are explicit on the API to use 2. we could have flexibility in the future to force run_instances API also with InstancTypeList (with one intance) 3. we don't propagate this overloaded mechanism to specify instance types to internal modules

Reply:
Added `API` field with this patch: https://github.com/aws/aws-parallelcluster-cookbook/pull/1508